### PR TITLE
dd-26-move-jest

### PR DESCRIPTION
**Pull request by Dewlar**
+ remove karma & jasmine
+ add jest
+ add @planess/train-a-backend library  

image

### CODE REVIEW

File: jest.config.ts
Comment (RykovaMariia) [inline jest.config.ts:1]: can we remove unnecessary comments?

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  // unmockedModulePathPatterns: undefined,
+
+  // Indicates whether each individual test should be reported during the run
+  // verbose: undefined,
+
+  // An array of regexp patterns that are matched against all source file paths before re-running tests in watch mode
+  // watchPathIgnorePatterns: [],
+
+  // Whether to use watchman for file crawling
+  // watchman: true,
```
Comment (RykovaMariia) [inline jest.config.ts]: don't we need this configuration for paths in tests?
I've had mistakes if I didn't have one.

```suggestion
 moduleNameMapper: {
    '^@core/(.*)$': '<rootDir>/src/app/core/$1',
    '^@features/(.*)$': '<rootDir>/src/app/features/$1',
    '^@shared/(.*)$': '<rootDir>/src/app/shared/$1',
  },
```

File: jest.config.ts
Comment (Dewlar) [reply jest.config.ts:1]: Yes, sure.
This is the default configuration. And I don’t know which of these settings may be useful to us.

Could you please suggest your version of the configuration file for jest?

Comment (Dewlar) [reply jest.config.ts:1]: should I remove all commented out settings?

Comment (RykovaMariia) [reply jest.config.ts:1]: I suggest deleting everything that is commented out, the rest is ok with me

File: jest.config.ts (hunk 1 above)
Comment (Dewlar) [reply jest.config.ts]: added

File: jest.config.ts
Comment (Dewlar) [reply jest.config.ts:1]: so it's cleaned up

**Review by YuliyaVoronovich — APPROVED** [review]

**Review by RykovaMariia — APPROVED** [review]

**Review by ChickenMcNuggetss — APPROVED** [review]

**Pull request merged**